CDS/scripts/js_utils/datastore/db.js

```diff
@@ -0,0 +1,120 @@
+const sqlite3 = require('sqlite3');
+
+class Database {
+    constructor (whoami, datastoreLocation = ':memory:') {
+        this.db = new sqlite3.Database(datastoreLocation);
+
+        this.whoami = whoami;
+
+        this.recordTypes = [ 'meta', 'media', 'tracks' ];
+    }
+
+    close () {
+        this.db.close();
+    }
+
+    getOne (type, key) {
+        return new Promise((resolve, reject) => {
+            if (! this.recordTypes.includes(type)) {
+                reject(`type must be ${this.recordTypes.join(', ')}`);
+            }
+
+            this._getOrCreateSource(type).then(() => {
+                this.db.serialize(() => {
+                    const sql = `
+                        SELECT  value 
+                        FROM    ${type} 
+                        WHERE   key = ?;`;
+
+                    this.db.prepare(sql).get(key, (err, row) => {
+                        if (err) {
+                            reject(err);
+                        }
+
+                        const value = row && row.value;
+                        const response = { value: value, type: type, key: key };
+
+                        resolve(response);
+                    });
+
+                });
+            });
+        });
+    }
+
+    setOne (type, key, value) {
+        return this.setMany(type, {[key]: value})
+    }
+
+    setMany (type, meta) {
+        return new Promise((resolve, reject) => {
+            if (! this.recordTypes.includes(type)) {
+                reject(`type must be ${this.recordTypes.join(', ')}`);
+            }
+
+            this._getOrCreateSource(type).then(sourceId => {
+                const sql = `
+                    INSERT INTO ${type} 
+                    (source_id, key, value) 
+                    VALUES (?, ?, ?);`;
+
+                const promises = Object.keys(meta).map(key => new Promise((resolve, reject) => {
+                    const values = [sourceId, key, meta[key]];
+                    this.db.prepare(sql).run(values, (err) => ! err ? resolve() : reject(err));
+                }));
+
+                Promise.all(promises)
+                    .then(values => resolve())
+                    .catch(err => reject(err));
+            });
+        });
+    }
+
+    _getOrCreateSource (type) {
+        return new Promise((resolve, reject) => {
+            this._getSource(type)
+                .then(id => resolve(id))
+                .catch(() => {
+                    this._createSource(type)
```

Comment:
Yeah?? (I'm not sure) Its the [current behaviour](https://github.com/guardian/content_delivery_system/blob/master/CDS/scripts/js_utils/Datastore.js#L60-L71) at least...